android/src/main/kotlin/dev/msfjarvis/claw/android/ui/ext.kt

```diff
@@ -80,6 +81,56 @@ fun PostActions(
   }
 }
 
+fun TwoPaneLayoutPostActions(
+  context: Context,
+  urlLauncher: UrlLauncher,
+  viewModel: ClawViewModel,
+  selectedPostId: MutableState<String?>,
+): PostActions {
+  return object : PostActions {
+    override fun viewPost(postId: String, postUrl: String, commentsUrl: String) {
+      viewModel.markPostAsRead(postId)
+      urlLauncher.openUri(postUrl.ifEmpty { commentsUrl })
+    }
+
+    override fun viewComments(postId: String) {
+      viewModel.markPostAsRead(postId)
+      selectedPostId.value = postId // Update selectedPostId using .value
+    }
```

Comment:
Hi @msfjarvis! The detailed pane in the TwoPaneLayout now displays the Comments Page, but this only seems to work for the first post that a user clicks on. Every subsequent post that is clicked just displays a split screen where the lower half says "No Comment", while still displaying the title from the previously clicked post. Is there something in the ViewModel that needs to be amended to resolve this issue?

Reply:
I've suggested the changes that should resolve it, the PR is once again missing a whole file so I can't test it myself.